### ISSUE TITLE
DialogHost not showing Singleton View

### ISSUE THREAD

**Issue by Diyari-Kurdi** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3126]
### Bug explanation

Hello! I'm using dependency injection and have registered a Usercontrol as a **Singleton**. The issue I'm facing is that when I open the DialogHost with the Usercontrol as its content, it works for the first time. However, when I repeat the process on a different window and close the new window, the Usercontrol stops showing up on the first window.
### [Demo.zip](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/files/10916965/Demo.zip)
https://user-images.githubusercontent.com/65598554/223613186-8b46d24a-9db4-4aeb-89c4-ee14e98c9b74.mp4

### Version

4.8

**Comment by nicolaihenriksen** [comment]
@Diyari-Kurdi I don't believe this is an MDIX issue in particular; more a general WPF thing. Registering a `UserControl` as a singleton and using it multiple places seems like a risky approach to me. My concerns here are about connecting/disconnecting elements in the visual tree when you "move the user control around".

Judging from your video, I am guessing you want it as a singleton to preserve the state across all your `DialogHost` instances? If that is the case, consider registering a singleton ViewModel instead (e.g. `MyUserControlViewModel`) which can hold the shared state, and then use a `DataTemplate` to visualize the view model in each `DialogHost`. Conceptually this gives you a new instance (possibly cached) of the `UserControl` every time a `DialogHost` has its dialog shown, but they will all deal with the same underlying ViewModel which will be automatically set as the `UserControl.DataContext`. Something along these lines (untested code!):

```xaml
<ContentControl Content="{Binding ...your singleton view model...}">
  <ContentControl.Resources>
    <DataTemplate DataType="{x:Type local:MyUserControlViewModel}">
      <local:MyUserControl />
    </DataTemplate>
  </ContentControl.Resources>
</ContentControl>
```

**Comment by Diyari-Kurdi** [comment]
@nicolaihenriksen Thank you for your comment! The actual problem is not with the data (ViewModel), but rather to the visual elements. For Example, if a UserControl have some heavy UI Elements, it takes some time to open the DialogHost each time. That's why I've registered it as a Singleton. However, you are correct in saying that I need to disconnect the UserControl when the DialogHost is closing. Thanks again!
```
private void DialogClosing(object sender, DialogClosingEventArgs eventArgs)
{
    MainDialogHost.DialogContent = null;
}
```
